src/main/java/de/dennisguse/opentracks/util/PreferencesUtils.java

```diff
@@ -318,4 +345,76 @@ public static void setDefaultExportDirectoryUri(SharedPreferences sharedPreferen
     public static boolean isDefaultExportDirectoryUri(SharedPreferences sharedPreferences, Context context) {
         return getDefaultExportDirectoryUri(sharedPreferences, context) != null;
     }
+
+    public static int getLayoutColumns(SharedPreferences sharedPreferences, Context context) {
+        return getInt(sharedPreferences, context, R.string.stats_custom_layout_columns_key, context.getResources().getInteger(R.integer.stats_custom_layout_columns_default));
+    }
+
+    public static void setLayoutColumns(SharedPreferences sharedPreferences, Context context, int columns) {
+        setInt(sharedPreferences, context, R.string.stats_custom_layout_columns_key, columns);
+    }
+
+    private static List<TypedArray> getMultiTypedArray(Context context, String key) {
+        List<TypedArray> typedArrays = new ArrayList<>();
+
+        try {
+            Class<R.array> resource = R.array.class;
```

Comment:
This looks complicated; what is happening here?
This feels like reflection...

Is there some nicer approach?
Using switch statement?

Reply:
It gets all array resources. I mean, `resource` has all array resources.

My first approach was using this:
`int resId = context.getResources().getIdentifier(key + "_" + counter, "array", context.getPackageName());`